### PR TITLE
another try

### PR DESCRIPTION
**Pull request by yevgenypats**
Giving a try on top of @disq PR.

Seems there are number of issues that Im still unsure why they happen:

1) Does the zero copy really works and produce correct results?
2) What's up with the timestamp read - seems something is messed up when we read back from parquet? maybe it's because of the list things?

### CODE REVIEW

File: parquet/read.go [hunk 1]
```diff
@@ -188,12 +189,10 @@ func reverseTransformInet(col *array.String) arrow.Array {
 func reverseTransformUUID(col *array.String) arrow.Array {
 	bldr := types.NewUUIDBuilder(array.NewExtensionBuilder(memory.DefaultAllocator, types.ExtensionTypes.UUID))
 	for i := 0; i < col.Len(); i++ {
-		if col.IsNull(i) {
+		if !col.IsValid(i) || col.Value(i) == "" {
```
Comment (yevgenypats) [inline parquet/read.go:192]: This made it work in some sense but I've no idea why this is needed. it Should be invalid.

File: parquet/write.go [hunk 2]
```diff
@@ -15,17 +15,21 @@ import (
 
 func (*Client) WriteTableBatch(w io.Writer, table *schema.Table, records []arrow.Record) error {
 	props := parquet.NewWriterProperties(
+		parquet.WithVersion(parquet.V2_4),
```
Comment (disq) [inline parquet/write.go:18]: shouldn't need this the default version is V2_LATEST

File: parquet/read.go [hunk 3]
```diff
@@ -81,41 +81,42 @@ func reverseTransformArray(f arrow.Field, col arrow.Array) arrow.Array {
 		return reverseTransformMAC(col.(*array.String))
 	case arrow.TypeEqual(dt, types.ExtensionTypes.JSON):
 		return reverseTransformJSON(col.(*array.String))
-	case arrow.TypeEqual(col.DataType(), arrow.FixedWidthTypes.Timestamp_ms):
+	case arrow.TypeEqual(dt, arrow.FixedWidthTypes.Timestamp_ns) || arrow.TypeEqual(dt, arrow.FixedWidthTypes.Timestamp_s):
```
Comment (disq) [inline parquet/read.go:84]: timestamps were discussed in the older PR: it somehow persists everything as ms I think... for sure it coerces all seconds to ms (even though timestamp coercion is off by default), but other timestamp units should remain as-is (but they don't?).

File: parquet/write.go (hunk 2 above)
Comment (yevgenypats) [reply parquet/write.go:18]: I was trying different options to see if anything comes up.

File: parquet/read.go (hunk 3 above)
Comment (yevgenypats) [reply parquet/read.go:84]: Yeah exactly. If it's a bug we should prob fix it in arrow I was mostly just trying to get the tests to pass to see where they fail next.

File: parquet/read.go (hunk 1 above)
Comment (disq) [reply parquet/read.go:192]: yes I think we have an issue persisting the null bitmap while writing.

**Pull request closed without merging**